vaadin-spring/src/test/java/com/vaadin/flow/spring/security/stateless/JwtSecurityContextRepositoryTest.java

```diff
@@ -0,0 +1,797 @@
+        JWTClaimsSet claimsSet = getClaimsSetBuilder().build();
+        Mockito.doReturn(getJwtAuthenticationToken(header, claimsSet))
+                .when(securityContext).getAuthentication();
+        final ImmutableSecret secret = new ImmutableSecret<>(TEST_OTHER_KEY);
+        jwtSecurityContextRepository.setJwkSource(secret);
+
+        jwtSecurityContextRepository.saveContext(securityContext, request,
+                response);
+
+        ArgumentCaptor<String> jwtCaptor = ArgumentCaptor
+                .forClass(String.class);
+        Mockito.verify(serializedJwtSplitCookieRepository).saveSerializedJwt(
+                jwtCaptor.capture(), ArgumentMatchers.eq(request),
+                ArgumentMatchers.eq(response));
+        String serializedJwt = jwtCaptor.getValue();
+        Assert.assertNotNull(serializedJwt);
+        Assert.assertNotEquals("", serializedJwt);
+        jwtProcessor.setJWSKeySelector(
+                new JWSVerificationKeySelector<>(JWSAlgorithm.HS256, secret));
+        JWTClaimsSet decodedClaimsSet = jwtProcessor.process(serializedJwt,
+                null);
+        assertClaims(decodedClaimsSet, TEST_USERNAME, TEST_ROLES, 1800);
+    }
+
+    @Test
+    public void saveContext_doesSaveJwt_withAlgoritm()
+            throws JOSEException, BadJOSEException, ParseException {
+        SecurityContext securityContext = SecurityContextHolder
+                .createEmptyContext();
+        JWSHeader header = getHeaderBuilder().build();
+        JWTClaimsSet claimsSet = getClaimsSetBuilder().build();
+        Mockito.doReturn(getJwtAuthenticationToken(header, claimsSet))
+                .when(securityContext).getAuthentication();
+        final ImmutableSecret secret = new ImmutableSecret<>(
+                TEST_64BYTE_KEY.getEncoded());
+        jwtSecurityContextRepository.setJwkSource(secret);
+        jwtSecurityContextRepository.setJwsAlgorithm(JWSAlgorithm.HS512);
+
+        jwtSecurityContextRepository.saveContext(securityContext, request,
+                response);
+
+        ArgumentCaptor<String> jwtCaptor = ArgumentCaptor
+                .forClass(String.class);
+        Mockito.verify(serializedJwtSplitCookieRepository).saveSerializedJwt(
+                jwtCaptor.capture(), ArgumentMatchers.eq(request),
+                ArgumentMatchers.eq(response));
+        String serializedJwt = jwtCaptor.getValue();
+        Assert.assertNotNull(serializedJwt);
+        Assert.assertNotEquals("", serializedJwt);
+        jwtProcessor.setJWSKeySelector(
+                new JWSVerificationKeySelector<>(JWSAlgorithm.HS512, secret));
+        JWTClaimsSet decodedClaimsSet = jwtProcessor.process(serializedJwt,
+                null);
+        assertClaims(decodedClaimsSet, TEST_USERNAME, TEST_ROLES, 1800);
+    }
+
+    private void assertEmptySecurityContext(SecurityContext securityContext) {
+        Mockito.verifyNoInteractions(securityContext);
+        assertRequestResponseHolder();
+    }
```

Comment:
There are a lot of tests which could be using this but are instead explicitly doing the exact same thing (2 lines) as this helper. See all test methods which have `_returnsEmptyContext_` in their name.

This is currently only used in one test `loadContext_returnsEmptyContext_when_nullJwt` and even there it additionally calls `assertRequestResponseHolder();` unnecessarily (since it's already covered by this helper).

Reply:
Cleaned up.